ext/BoundaryValueDiffEqODEInterfaceExt.jl

```diff
@@ -1,8 +1,13 @@
 module BoundaryValueDiffEqODEInterfaceExt
 
-using SciMLBase, BoundaryValueDiffEq, ODEInterface, RecursiveArrayTools
-import BoundaryValueDiffEq: __extract_u0, __flatten_initial_guess, __extract_mesh,
-                            __initial_guess_length, __initial_guess, __has_initial_guess
+using SciMLBase, BoundaryValueDiffEq, ODEInterface, RecursiveArrayTools, ConcreteStructs,
+      Setfield
+using PreallocationTools
+import SciMLBase: AbstractDiffEqInterpolation, StandardBVProblem, __solve, _unwrap_val
+include("../lib/BoundaryValueDiffEqCore/src/types.jl")
+include("../lib/BoundaryValueDiffEqCore/src/utils.jl")
+include("../lib/BoundaryValueDiffEqCore/src/algorithms.jl")
+include("../lib/BoundaryValueDiffEqCore/src/BoundaryValueDiffEqCore.jl")
```

Comment:
why is this all directly included?

Reply:
Some functionalities in wrapper solvers require BoundaryValueDiffEqCore, but cannot directly use it since the core is not registered or can be directly `import ..BoundaryValueDiffEqCore` in the extension, so just temporarily including the necessary functionalities

Reply:
okay as long as it's temporary